Tensile/KernelWriterAssembly.py

```diff
@@ -3646,16 +3649,36 @@ def graTileAssignment(self, kernel, tP):
           kStr += vectorStaticRemainder(wReg, tmpVgpr, num1DWaves, tmpSgpr)
           kStr += staticMultiply(vgpr(wReg), vgpr(wReg), strideWave, sgpr(tmpSgpr))
           kStr += inst("_v_add_u32", vgpr(rReg), vgpr(wReg), vgpr(rReg),"")
-          # need division for qReg
-          kStr += vectorStaticDivide(qReg, qReg, kernel["MatrixInstN"], tmpSgpr)
-          lrvwOther = self.lrvwB if tP["isA"] else self.lrvwA # The other side of lrvw
-          tluOther = kernel["ProblemType"]["TLUB"] if tP["isA"] else kernel["ProblemType"]["TLUA"] # The other side of tlu
-          if lrvwOther >= 2 and (not tluOther) and tP["tlu"]:
-            # DirectToVgpr + LocalReadVectorWidth>=2 case, multiply qReg by lrvwOther
-            kStr += staticMultiply(vgpr(qReg), vgpr(qReg), lrvwOther, sgpr(tmpSgpr))
           # release register
           self.vgprPool.checkIn(wReg)
           self.vgprPool.checkIn(tmpVgpr)
+
+      # need division for qReg
+      kStr += vectorStaticDivide(qReg, qReg, kernel["MatrixInstN"], tmpSgpr)
+
+      # localSplitU case. Calculate LSU offset here
+      if kernel["LocalSplitU"] > 1:
+          # alloc vgpr
+          tmpVgpr = self.vgprPool.checkOut(1, 'graTA vgpr', self.preventVgprOverflowDuringNewTile)
+          subGroup  = kernel["SubGroup0"] * kernel["SubGroup1"]
+          dividendForKId   = kernel["MatrixInstM"] * kernel["MatrixInstB"]
+          kInterval = waveWidth // dividendForKId
+          # generate instruction
+          kStr += vectorStaticDivide(tmpVgpr, "Serial", subGroup, tmpSgpr, \
+            "LSU offset: sgid = Serial / subGroup(%u)" % subGroup)
+          kStr += staticMultiply(vgpr(tmpVgpr), vgpr(tmpVgpr), kInterval, sgpr(tmpSgpr), \
+            "LSU offset: lsuOffset = sgid * kInterval(%u)" % (kInterval))
+          kStr += inst("_v_add_u32", vgpr(qReg), vgpr(tmpVgpr), vgpr(qReg), \
+            "LSU offset: lrOffset += lsuOffset")
+          # release register
+          self.vgprPool.checkIn(tmpVgpr)
```

Comment:
Is this the same as the code in LraTileAssignment? Is the offset saved for re-use?